Feature/5611 : Adding filters for featured image minimum height and width

naveen17797 opened the pull request:
## Summary

Fixes #5611

This adds a filter for featured image height and width, allowing external plugins to alter the value.

## Checklist

- [x] My pull request is addressing an open issue (please create one otherwise).
- [x] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [x] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

naveen17797 committed:
see #5611: [RED] test failing for minimum height and width keys

naveen17797 committed:
see #5611: [GREEN] test passing for minimum height and width keys

naveen17797 committed:
see #5611: [GREEN] test passing for featured image dimensions

naveen17797 committed:
see #5611: [GREEN] test passing for jest getFeaturedImage

naveen17797 committed:
see #5611: [BLUE] minor fix on jest test

naveen17797 committed:
see #5611: [BLUE] change test description

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/ampproject/amp-wp?pullRequest=5647) <br/>All committers have signed the CLA.

naveen17797 committed:
see #5611: [BLUE] fixing lint errors

naveen17797 commented:
This is ready for review @westonruter 

i ran the lint locally, i didnt find any errors for php and js, but here js lint seems to fail due to different error

`Error: Resource not accessible by integration`

and a unit test fails which as far as i know, its not related to the currently added feature, how should i proceed ?

westonruter commented:
The unit test failure will be fixed if you merge the latest from `develop`. I fixed that test failure in #5654.

Please sign the CLA: https://github.com/ampproject/amp-wp/pull/5647#issuecomment-737921279.

As for the lint error, it's probably a problem with the GitHub Action being run on a fork and not on an internal PR. @pierlon is that the case?

pierlon commented:
> As for the lint error, it's probably a problem with the GitHub Action being run on a fork and not on an internal PR. @pierlon is that the case?

@westonruter Yes that is the case here. The Action doesn't have the necessary permissions to annotate the pull request diff when run from a fork. We could disable the annotation step on PR forks for now, and then look for any alternatives that could resolve this limitation.

westonruter commented:
@pierlon Good. Please do that and then @naveen17797 can update from `develop`.

pierlon commented:
I've created https://github.com/ampproject/amp-wp/pull/5656 that will skip the annotation step when run from a PR fork.

westonruter commented:
PR merged. @naveen17797 Please update your PR with the latest from `develop`.

naveen17797 committed:
Merge remote-tracking branch 'upstream/develop' into feature/5611-featured-image-height-width

naveen17797 commented:
i have updated from develop and signed the CLA.

westonruter commented on includes/admin/class-amp-post-meta-box.php:
```diff
@@ -292,6 +292,40 @@ function () {
+			 *
+			 * @return int
+			 */
+			'featuredImageMinimumHeight' => apply_filters( 'amp_featured_image_minimum_height', 675 ),
```
```suggestion
			'featuredImageMinimumHeight' => (int) apply_filters( 'amp_featured_image_minimum_height', 675 ),
```

westonruter commented on includes/admin/class-amp-post-meta-box.php:
```diff
@@ -292,6 +292,40 @@ function () {
+			 *
+			 * @return int
+			 */
+			'featuredImageMinimumWidth'  => apply_filters( 'amp_featured_image_minimum_width', 1200 ),
```
```suggestion
			'featuredImageMinimumWidth'  => (int) apply_filters( 'amp_featured_image_minimum_width', 1200 ),
```

westonruter commented on includes/admin/class-amp-post-meta-box.php:
```diff
@@ -292,6 +292,40 @@ function () {
+	 * Returns the featured image dimensions after filtering
+	 * the height and width
+	 *
+	 * @return array
```
```suggestion
	 * @return int[]
```

westonruter commented on assets/src/common/helpers/index.js:
```diff
@@ -55,9 +54,25 @@ export const hasMinimumDimensions = ( media, dimensions ) => {
+		window.ampBlockEditor.featuredImageMinimumHeight === undefined ) {
+		return defaultDimensions;
+	}
+	const height = parseInt( window.ampBlockEditor.featuredImageMinimumHeight );
+	const width = parseInt( window.ampBlockEditor.featuredImageMinimumWidth );
```
If using `(int)` casting per my suggestion comment below, then using `parseInt()` may not be relevant here.

naveen17797 replied:
we would still need parseInt, because wp_localize_script converts integers to string

![Screenshot from 2020-12-07 09-20-19](https://user-images.githubusercontent.com/18109258/101307964-b6d67c80-386e-11eb-9f9f-e7fe3b786af8.png)

Ticket:  https://core.trac.wordpress.org/ticket/25280

westonruter commented on includes/admin/class-amp-post-meta-box.php:
```diff
@@ -292,6 +292,40 @@ function () {
+	public static function get_featured_image_dimensions() {
+		return [
+			/**
+			 * Filter name: amp_featured_image_minimum_height
```
```suggestion
			 * Filters the minimum height required for a featured image.
```

westonruter commented on includes/admin/class-amp-post-meta-box.php:
```diff
@@ -292,6 +292,40 @@ function () {
+			'featuredImageMinimumHeight' => apply_filters( 'amp_featured_image_minimum_height', 675 ),
+
+			/**
+			 * Filter name: amp_featured_image_minimum_width
```
```suggestion
			 * Filters the minimum width required for a featured image.
```

westonruter commented on includes/admin/class-amp-post-meta-box.php:
```diff
@@ -292,6 +292,40 @@ function () {
+			 * @since 2.0.8
+			 *
+			 * @param int $featured_image_minimum_height The minimum width of the image,
+			 * defaults to 1200
```
```suggestion
			 * defaults to 1200.
```

westonruter commented on includes/admin/class-amp-post-meta-box.php:
```diff
@@ -292,6 +292,40 @@ function () {
+			 * @since 2.0.8
+			 *
+			 * @param int $featured_image_minimum_height The minimum height of the image,
+			 * defaults to 675
```
```suggestion
			 * defaults to 675.
```

westonruter commented on includes/admin/class-amp-post-meta-box.php:
```diff
@@ -292,6 +292,40 @@ function () {
+			 *
+			 * @param int $featured_image_minimum_height The minimum height of the image,
+			 * defaults to 675
+			 *
+			 * @return int
```
```suggestion
```

Filter phpdoc does not include `@return`. See https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#4-hooks-actions-and-filters

westonruter commented on includes/admin/class-amp-post-meta-box.php:
```diff
@@ -292,6 +292,40 @@ function () {
+			 *
+			 * @param int $featured_image_minimum_height The minimum width of the image,
+			 * defaults to 1200
+			 *
+			 * @return int
```
```suggestion
```

johnwatkins0 commented on assets/src/common/helpers/index.js:
```diff
@@ -55,9 +54,25 @@ export const hasMinimumDimensions = ( media, dimensions ) => {
+		return defaultDimensions;
+	}
+	const height = parseInt( window.ampBlockEditor.featuredImageMinimumHeight );
+	const width = parseInt( window.ampBlockEditor.featuredImageMinimumWidth );
```
This function could also be done a little more concisely with [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) (note the `?`s):

```JS
return {
	width: window.ampBlockEditor?.featuredImageMinimumWidth || 1200,
	height: window.ampBlockEditor?.featuredImageMinimumHeight || 675,
}
```

naveen17797 committed:
see #5611: Apply suggestions from code review

Co-authored-by: Weston Ruter <westonruter@google.com>

naveen17797 committed:
see #5611: modifying featured image function based on review suggestion

naveen17797 committed:
see #5611: adding additional tests

naveen17797 committed:
see #5611: adding additional test to fix the coverage error

naveen17797 commented:
Hello @westonruter, i made the changes based on review. But i see the coverage test is failing which is little strange because i am sure i have added all the test cases to jest, i could not find any info on how to resolve this issue.

westonruter commented on includes/admin/class-amp-post-meta-box.php:
```diff
@@ -270,7 +270,7 @@ function () {
 			'hasThemeSupport' => ! amp_is_legacy(),
 			'isStandardMode'  => amp_is_canonical(),
-		];
+		] + self::get_featured_image_dimensions();
```
This should rather use `array_merge()`.

naveen17797 replied:
I have removed  `+` and replaced it with array_merge function

naveen17797 committed:
see #5611: replace + with array_merge function

pierlon committed:
Add tests for validateFeaturedImage function

pierlon commented:
> But i see the coverage test is failing which is little strange because i am sure i have added all the test cases to jest, i could not find any info on how to resolve this issue.

@naveen17797 I've added some tests in https://github.com/ampproject/amp-wp/pull/5647/commits/2b8e378865c53a09fcf67fdcc425030ba18311aa to increase the JS coverage.

naveen17797 commented:
> > But i see the coverage test is failing which is little strange because i am sure i have added all the test cases to jest, i could not find any info on how to resolve this issue.
> 
> @naveen17797 I've added some tests in [2b8e378](https://github.com/ampproject/amp-wp/commit/2b8e378865c53a09fcf67fdcc425030ba18311aa) to increase the JS coverage.

Thanks @pierlon

westonruter commented on assets/src/common/helpers/index.js:
```diff
@@ -55,11 +54,10 @@ export const hasMinimumDimensions = ( media, dimensions ) => {
-
-	return { width, height };
+	return {
+		width: parseInt( window?.ampBlockEditor?.featuredImageMinimumWidth ) || 1200,
+		height: parseInt( window?.ampBlockEditor?.featuredImageMinimumHeight ) || 675,
```
Come to think of it, wouldn't it be better to define `ampBlockEditor` as an external as is being done for the settings screen and the block editor? It's actually already marked as one:

https://github.com/ampproject/amp-wp/blob/85f53e68341b784da35b3bd2661cdbbdf6f20f63/webpack.config.js#L94-L97

So then we could use it like so:

https://github.com/ampproject/amp-wp/blob/85f53e68341b784da35b3bd2661cdbbdf6f20f63/assets/src/settings-page/index.js#L5-L10

Then this file could have at the top:

```js
import {
	featuredImageMinimumWidth,
	featuredImageMinimumHeight,
} from 'amp-block-editor';
```

And then we can use it here. 

We can eliminate the hard-coding of `1200` and `675` in the JS file by forcing the `get_featured_image_dimensions` method to always return two valid dimensions.

naveen17797 replied:
so i need to check if both the height and width are valid integers in the server side and then pass it to js ?

westonruter replied:
Yeah, what about this: https://github.com/ampproject/amp-wp/pull/5647#pullrequestreview-547518534

westonruter commented on includes/admin/class-amp-post-meta-box.php:
```diff
@@ -292,6 +295,36 @@ function () {
+	 * @return int[]
+	 */
+	public static function get_featured_image_dimensions() {
+		return [
```
```suggestion
		$default_width  = 1200;
		$default_height = 675;

		$dimensions = [
```

westonruter commented on includes/admin/class-amp-post-meta-box.php:
```diff
@@ -292,6 +295,36 @@ function () {
+			 * @param int $featured_image_minimum_height The minimum height of the image,
+			 * defaults to 675.
+			 */
+			'featuredImageMinimumHeight' => (int) apply_filters( 'amp_featured_image_minimum_height', 675 ),
```
```suggestion
			'featuredImageMinimumHeight' => (int) apply_filters( 'amp_featured_image_minimum_height', $default_height ),
```

westonruter commented on includes/admin/class-amp-post-meta-box.php:
```diff
@@ -292,6 +295,36 @@ function () {
+			 * @param int $featured_image_minimum_height The minimum width of the image,
+			 * defaults to 1200.
+			 */
+			'featuredImageMinimumWidth'  => (int) apply_filters( 'amp_featured_image_minimum_width', 1200 ),
```
```suggestion
			'featuredImageMinimumWidth'  => (int) apply_filters( 'amp_featured_image_minimum_width', $default_width ),
```

westonruter commented on includes/admin/class-amp-post-meta-box.php:
```diff
@@ -292,6 +295,36 @@ function () {
+			 * defaults to 1200.
+			 */
+			'featuredImageMinimumWidth'  => (int) apply_filters( 'amp_featured_image_minimum_width', 1200 ),
+		];
```
```suggestion
		];
		if ( $dimensions['featuredImageMinimumHeight'] <= 0 ) {
			$dimensions['featuredImageMinimumHeight'] = $default_height;
		}
		if ( $dimensions['featuredImageMinimumWidth'] <= 0 ) {
			$dimensions['featuredImageMinimumWidth'] = $default_width;
		}
		return $dimensions;
```

naveen17797 committed:
Apply suggestions from code review

Co-authored-by: Weston Ruter <westonruter@google.com>

naveen17797 committed:
see #5611: load featured image height and width using externals, add mocks

naveen17797 commented:
@westonruter I have added the changes, loaded featuredImageHeight, width keys using externals.

naveen17797 commented:
And a test configuration fails due to network error

```
{"offers":[{"response":"upgrade","download":"http:\/\/downloads.wordpress.org\/release\/wordpress-5.6.zip","locale":"en_US","packages":{"full":"http:\/\/downloads.wordpress.org\/release\/wordpress-5.6.zip","no_content":"http:\/\/downloads.wordpress.org\/release\/wordpress-5.6-no-content.zip","new_bundled":"http:\/\/downloads.wordpress.org\/release\/wordpress-5.6-new-bundled.zip","partial":false,"rollback":false},"current":"5.6","version":"5.6","php_version":"5.6.20","mysql_version":"5.0","new_bundled":"5.6","partial_version":false}],"translations":[]}
Installing WP from https://develop.svn.wordpress.org/tags/5.6/ to /tmp/wordpress
svn: E000104: Error running context: Connection reset by peer
```

westonruter commented:
Must have been a temporary failure due to heavy server load for WordPress 5.6 downloads.

westonruter commented on assets/src/common/helpers/index.js:
```diff
@@ -55,11 +55,10 @@ export const hasMinimumDimensions = ( media, dimensions ) => {
-
-	return { width, height };
+	return {
+		width: parseInt( featuredImageMinimumWidth ),
+		height: parseInt( featuredImageMinimumHeight ),
```
The use of `parseInt()` seems obsolete now.

```suggestion
		width: featuredImageMinimumWidth,
		height: featuredImageMinimumHeight,
```

naveen17797 replied:
@westonruter wp_localize_script will provide it as a string even though we supply integer to it, we still need to convert it right ?

westonruter replied:
Oh yes, you're right. Please then also change this:

https://github.com/ampproject/amp-wp/blob/a1a7e8cb7d085bd06f73fb1d25cd300670dc0c22/includes/admin/class-amp-post-meta-box.php#L275-L279

To:

```php
wp_add_inline_script(
	self::BLOCK_ASSET_HANDLE,
	sprintf( 'var ampBlockEditor = %s;', wp_json_encode( $data ) ),
	'before'
);
```

I consider this to be a best practice for exporting PHP data to JS instead of using `wp_localize_script()`.

naveen17797 replied:
Done, i will fix the tests.

naveen17797 committed:
see #5611: replace wp localize script with inline script

naveen17797 committed:
see #5611: fix tests

westonruter commented on includes/admin/class-amp-post-meta-box.php:
```diff
@@ -292,6 +295,46 @@ function () {
+			 *
+			 * @since 2.0.9
+			 *
+			 * @param int $featured_image_minimum_height The minimum height of the image,
+			 * defaults to 675.
```
```suggestion
			 * @param int $featured_image_minimum_height The minimum height of the image,
			 *                                           defaults to 675.
```

westonruter commented on includes/admin/class-amp-post-meta-box.php:
```diff
@@ -292,6 +295,46 @@ function () {
+			 *
+			 * @since 2.0.9
+			 *
+			 * @param int $featured_image_minimum_height The minimum width of the image,
+			 * defaults to 1200.
```
```suggestion
			 * @param int $featured_image_minimum_height The minimum width of the image,
			 *                                           defaults to 1200.
```

westonruter commented on tests/php/test-class-amp-meta-box.php:
```diff
@@ -158,20 +158,34 @@ public function test_enqueue_block_assets() {
+		$this->assertArrayHasKey( 'featuredImageMinimumHeight', $dimensions );
+		$this->assertArrayHasKey( 'featuredImageMinimumWidth', $dimensions );
+		$this->assertEquals( $dimensions['featuredImageMinimumWidth'], 1200 );
+		$this->assertEquals( $dimensions['featuredImageMinimumHeight'], 675 );
```
Please also add `amp_featured_image_minimum_width` and `amp_featured_image_minimum_height` filters, first asserting that adding >0 numbers get returned when calling `AMP_Post_Meta_Box::get_featured_image_dimensions()`, and then that again that when ≤0 numbers are returned by these filters that `AMP_Post_Meta_Box::get_featured_image_dimensions()` returns the defaults.

westonruter reviewed (changes requested):
Almost there!

naveen17797 committed:
Apply suggestions from code review

Co-authored-by: Weston Ruter <westonruter@google.com>

naveen17797 committed:
see #5611: adding tests for valid usecases

naveen17797 committed:
Merge remote-tracking branch 'origin/feature/5611-featured-image-height-width' into feature/5611-featured-image-height-width

naveen17797 committed:
see #5611: fix tests

westonruter committed:
Use coversDefaultClass and covers tags; improve assertEquals arg order

westonruter commented on includes/admin/class-amp-post-meta-box.php:
```diff
@@ -292,6 +295,46 @@ function () {
+			 */
+			'featuredImageMinimumWidth'  => (int) apply_filters( 'amp_featured_image_minimum_width', $default_width ),
+		];
+		if ( $dimensions['featuredImageMinimumHeight'] <= 0 ) {
+			$dimensions['featuredImageMinimumHeight'] = $default_height;
+		}
+		if ( $dimensions['featuredImageMinimumWidth'] <= 0 ) {
+			$dimensions['featuredImageMinimumWidth'] = $default_width;
+		}
```
Something that comes to mind: would there be value to be able to filter `amp_featured_image_minimum_height` and/or `amp_featured_image_minimum_width` to be `-1` or `false` in order to disable constraints in one dimension, or to eliminate the constraints altogether if they are both filtered to be `-1` or `false`? I realize this contradicts what I suggested where that method always has to return valid integer dimensions.  In the case of the value being less than or equal to `0`, then the value of `false` could be supplied instead of the defaults. Then the JS would have to account for the values being either integers or `false` and update the UI accordingly. 

Would this be useful? 

Perhaps a site owner would like to disable the featured image dimension constraints altogether, and this would provide them a way to do this. cc @pierlon @johnwatkins0

westonruter replied:
In other words, the question is whether we could actually allow a site author to do:

```php
add_filter( 'amp_featured_image_minimum_height', '__return_false' );
```

or

```php
add_filter( 'amp_featured_image_minimum_width', '__return_false' );
```

or

```php
add_filter( 'amp_featured_image_minimum_width', '__return_false' );
add_filter( 'amp_featured_image_minimum_height', '__return_false' );
```

pierlon replied:
I'd say it's highly unlikely a site owner would want to disable constraints for one dimension. If the dimensions are filtered to a height & width of `0` (or maybe `false`), then I'd consider the constraints to be disabled.

westonruter replied:
You're right. In any case, it seems unsymmetric to allow filtering _either_ of the dimensions to disable the constraints. I've pushed up what I had in mind.

If I do:

```php
add_filter( 'amp_featured_image_minimum_width', '__return_false' );
```

I get:

Media Library | Document Sidebar | Pre-publish Check
--------------|--------------------|-------------------
![image](https://user-images.githubusercontent.com/134745/102274710-28e73b00-3ed9-11eb-8773-16f1c86cbf42.png) | ![image](https://user-images.githubusercontent.com/134745/102274664-1a008880-3ed9-11eb-9d50-45edd4212af2.png) | ![image](https://user-images.githubusercontent.com/134745/102274727-313f7600-3ed9-11eb-8f6d-62b7ed35a77a.png)

 But if I instead do:

```php
add_filter( 'amp_featured_image_minimum_height', '__return_false' );
```

I get:

Media Library | Document Sidebar | Pre-publish Check
--------------|--------------------|-------------------
![image](https://user-images.githubusercontent.com/134745/102274789-487e6380-3ed9-11eb-9495-81230744382c.png) | ![image](https://user-images.githubusercontent.com/134745/102274836-5a600680-3ed9-11eb-8ade-18d2af6b02e1.png) | ![image](https://user-images.githubusercontent.com/134745/102274843-5fbd5100-3ed9-11eb-9c47-c56ec50a79c6.png)

And if I do:

```php
add_filter( 'amp_featured_image_minimum_width', '__return_false' );
add_filter( 'amp_featured_image_minimum_height', '__return_false' );
```

I get:

Media Library | Document Sidebar | Pre-publish Check
--------------|--------------------|-------------------
![image](https://user-images.githubusercontent.com/134745/102274904-7663a800-3ed9-11eb-8a9e-ba02ba6eab8d.png) | ![image](https://user-images.githubusercontent.com/134745/102274919-7c598900-3ed9-11eb-8b9d-57369db93a94.png) | ![image](https://user-images.githubusercontent.com/134745/102274936-824f6a00-3ed9-11eb-8192-ee455efa6c10.png)

And if I have no such filters at all, then I get:

Media Library | Document Sidebar | Pre-publish Check
--------------|--------------------|-------------------
![image](https://user-images.githubusercontent.com/134745/102275442-410b8a00-3eda-11eb-9ee6-033e683cda73.png) | ![image](https://user-images.githubusercontent.com/134745/102275467-48329800-3eda-11eb-95b3-cb72be784446.png) | ![image](https://user-images.githubusercontent.com/134745/102275488-4f59a600-3eda-11eb-86d5-d402e459057c.png)

westonruter replied:
@naveen17797 What do you think of this? I don't believe it impacts your use case, but would allow others to disable the checks entirely.

naveen17797 replied:
@westonruter My point of view would be adding a separate filter for enabling / disabling the featured image check like

add_filter( 'amp_featured_image_check', '__return_false' );

so that we will keep the concerns separate ( height, width filters can only be used to alter the dimensions, and separate filter to enable/disable the check ),  may be i am wrong.. 

what do you think ?

westonruter replied:
That would work too, but since we already have the `amp_featured_image_minimum_width` and `amp_featured_image_minimum_height` filters, it seems unnecessary to add yet another filter if those two can have the same effect by filtering the values to be `0`.

naveen17797 replied:
> it seems unnecessary to add yet another filter

correct, so we can disable the check if the height or width value is zero ?

westonruter replied:
If _both_ are filtered to be zero (or `false`):

```php
add_filter( 'amp_featured_image_minimum_width', '__return_false' );
add_filter( 'amp_featured_image_minimum_height', '__return_false' );
```

naveen17797 replied:
ok, should i need to open a separate issue for this ?

westonruter replied:
For what? I implemented this already in this PR. See my screenshots above: https://github.com/ampproject/amp-wp/pull/5647#discussion_r543700197.

naveen17797 replied:
i think this needs to be conditional ( only add the filter when the check needs to be done ? ).

`addFilter( 'editor.PostFeaturedImage', 'ampEditorBlocks/withFeaturedImageNotice', withFeaturedImageNotice );`

westonruter replied:
That's true. That logic could be omitted entirely if both are filtered to be false/zero. It's not only there, but it would also need to be for the two other contexts shown in my screenshots above. In any case, it doesn't harm anything to have the logic in place: it just doesn't do anything.

westonruter committed:
Allow filtering dimensions to be zero to disable constraints

Reuse hasMinimumDimensions() in FeaturedImageToolbarSelect

westonruter committed:
Avoid camelCase array keys in PHP

johnwatkins0 commented on assets/src/common/components/select-media-frame.js:
```diff
@@ -27,15 +27,36 @@ const NOTICE_CLASSNAME = 'notice notice-warning notice-alt inline';
-			__( 'The selected image is too small (%1$s by %2$s pixels). It should have a size of at least %3$s by %4$s pixels.', 'amp' ),
+		let message = sprintf(
+			/* translators: 1: image width in pixels. 2: image height in pixels. */
+			__( 'The selected image is too small (%1$s by %2$s pixels).', 'amp' ),
```
This `sprintf` now has two too many arguments.

johnwatkins0 commented on assets/src/common/helpers/test/__mocks__/amp-block-editor-data.js:
```diff
@@ -0,0 +1,11 @@
+ *
+ * @type {{featuredImageMinimumHeight: number, featuredImageMinimumWidth: number}}
+ */
+const mockAmpBlockEditorData = {
+	featuredImageMinimumWidth: 1200,
+	featuredImageMinimumHeight: 675,
+};
+
+module.exports = mockAmpBlockEditorData;
```
```suggestion
export const mockAmpBlockEditorData = {
	featuredImageMinimumWidth: 1200,
	featuredImageMinimumHeight: 675,
};

```

westonruter replied:
@johnwatkins0 this apparently did not work. See https://github.com/ampproject/amp-wp/pull/5647/checks?check_run_id=1561450704

johnwatkins0 replied:
Restored it to how it was e2b842f665ff0183475bf37fe5f2d4bf6d94bb35

Not sure what's going on, but it's too small worry about.

johnwatkins0 reviewed (changes requested):
A couple of little things. Looking good!

westonruter commented on assets/src/common/components/select-media-frame.js:
```diff
@@ -27,15 +27,36 @@ const NOTICE_CLASSNAME = 'notice notice-warning notice-alt inline';
+			__( 'The selected image is too small (%1$s by %2$s pixels).', 'amp' ),
 			'{{width}}',
 			'{{height}}',
 			'{{minWidth}}',
 			'{{minHeight}}',
```
@johnwatkins0 right, so like so?

```suggestion
```

johnwatkins0 replied:
Yup, looks right.

westonruter committed:
Improve exporting of constant

Co-authored-by: John Watkins <johnwatkins0@gmail.com>

johnwatkins0 commented on assets/src/common/helpers/test/__mocks__/amp-block-editor-data.js:
```diff
@@ -0,0 +1,9 @@
+ *
+ * @type {{featuredImageMinimumHeight: number, featuredImageMinimumWidth: number}}
+ */
+export const mockAmpBlockEditorData = {
+	featuredImageMinimumWidth: 1200,
+	featuredImageMinimumHeight: 675,
+};
```
Apologies, my last suggestion was mistaken. 

```suggestion
const mockAmpBlockEditorData = {
	featuredImageMinimumWidth: 1200,
	featuredImageMinimumHeight: 675,
};

export default mockAmpBlockEditorData;

```

westonruter committed:
Fix variable export

Co-authored-by: John Watkins <johnwatkins0@gmail.com>

westonruter committed:
Remove extra args from sprintf()

johnwatkins0 committed:
Restore mockAmpBlockEditorData to previous state

westonruter reviewed (approved):
This was a journey! Thanks for your patience and quality contributions!

westonruter committed:
Merge branch 'develop' of github.com:ampproject/amp-wp into feature/5611-featured-image-height-width

* 'develop' of github.com:ampproject/amp-wp: (102 commits)
  Use codeCoverageIgnore
  Make sure menu button container remains visible
  Persist focus on mobile menu button when toggling
  Unset wp_scripts and wp_styles after each test
  Add test for amend_twentytwentyone_styles
  Add test for amend_twentytwentyone_dark_mode_styles
  Remove !important rule
  Trap keyboard navigation within menu when it's open
  Replace amp-lightbox with an amp-bind solution
  Bump sirbrillig/phpcs-variable-analysis from 2.10.0 to 2.10.1
  Improve diff output for PHP files
  Try to maintain focus on button when opening and closing lightbox
  Use DOMElement instanceof check for static analysis
  Apply padding to spans within menu button toggle
  Use less specific XPath query to get menu toggles
  Add a warning notice to setting
  Make phpstan happy
  Use two discrete toggles for opening and closing mobile menu in Twenty Twenty-One
  Add a warning notice to setting
  Add force flag to rm on the productionVendorExcludedFilePatterns
  ...
